opencti-platform/opencti-front/src/utils/hooks/usePreloadedPaginationFragment.ts

```diff
@@ -0,0 +1,45 @@
+import { usePaginationFragment, usePreloadedQuery } from 'react-relay';
+import { useEffect } from 'react';
+import { PreloadedQuery } from 'react-relay/relay-hooks/EntryPointTypes';
+import { GraphQLTaggedNode, OperationType } from 'relay-runtime';
+import { KeyType } from 'react-relay/relay-hooks/helpers';
+import { UseLocalStorage } from './useLocalStorage';
+
+interface UsePreloadedPaginationFragment<QueryType extends OperationType> {
+  queryRef: PreloadedQuery<QueryType>
+  linesQuery: GraphQLTaggedNode
+  linesFragment: GraphQLTaggedNode
+  nodePath: string[]
+  setNumberOfElements?: UseLocalStorage[2]['handleSetNumberOfElements']
+}
+
+const usePreloadedPaginationFragment = <QueryType extends OperationType, FragmentKey extends KeyType>({
+  queryRef,
+  linesQuery,
+  linesFragment,
+  nodePath,
+  setNumberOfElements,
+}: UsePreloadedPaginationFragment<QueryType>) => {
+  const queryData = usePreloadedQuery(linesQuery, queryRef) as FragmentKey;
+  const {
+    data,
+    hasNext,
+    loadNext,
+    isLoadingNext,
+  } = usePaginationFragment<QueryType, FragmentKey>(linesFragment, queryData);
+  useEffect(() => {
+    const deep_value = nodePath.reduce((a, v) => a[v as keyof object], data as object) as unknown[];
+    if (setNumberOfElements && deep_value) {
+      setNumberOfElements({ number: deep_value.length });
```

Comment:
I don't understand why we do that

Reply:
Because the nb items of a list is shown in the Main component and the count is calculated in the Lines component. We need to share the info.
The useEffect is used to send the information everytime the list changes

Reply:
+1 for the explanation :)